### PR TITLE
Ls1 usherzhou fix

### PR DESCRIPTION
**Pull request by amartyads**
Fixes particle insertion and boundary forces with ls1. 2way coupling still needs some work but 1way works fine for now.

### CODE REVIEW

**Comment by amartyads** [comment]
I've added a lot of tests, and now the branch has both fixes and tests; let me know if the tests are okay

File: CMakeLists.txt [hunk 1]
```diff
@@ -301,7 +301,7 @@ if(BUILD_WITH_ADIOS2)
   target_compile_definitions(mamico INTERFACE BUILD_WITH_ADIOS2)
 endif()
 target_compile_definitions(mamico INTERFACE ${MD_SIM} MDDim3)
-target_compile_options(mamico INTERFACE -Wall -Wno-unknown-warning-option -Wno-unknown-pragmas -Wfatal-errors)
+target_compile_options(mamico INTERFACE -Wall -Wno-unknown-warning-option -Wno-unknown-pragmas -Wfatal-errors -Wno-reorder)
```
Comment (amartyads) [inline CMakeLists.txt:304]: The problem with this:
This is needed for AutoPas to compile, and I have fixed the issue in the AutoPas master branch. However ls1 does not use the latest AutoPas, it has to be updated to use the latest AutoPas, and the version used by ls1 is pretty outdated. Until ls1 is updated to use the newer versions, we need this flag to compile AutoPas. Do we keep the flag?

Comment (Thinkpiet) [reply CMakeLists.txt:304]: We keep the flag, but only if `if("${MD_SIM}" STREQUAL "LS1_MARDYN")`

**Comment by amartyads** [comment]
Few unit tests I couldn't finish in LS1MDSolverInterface because I didn't know what would be the best approach, would appreciate some help
* `getInitialVelocity()` (didn't know how to validate result)
* `setupPotentialEnergyLandscape` (empty function)
* `getLinkedCellIndexForMoleculePosition` and `calculateForceAndEnergy` (didn't know how to validate result without repeating logic in functions)

File: test/unit/coupling/interface/impl/ls1/LS1MoleculeIteratorTest.cpp
Comment (LouieVoit) [inline test/unit/coupling/interface/impl/ls1/LS1MoleculeIteratorTest.cpp:1]: Looks really good ! I validate.

**Comment by Thinkpiet** [comment]
Looks good but does not compile without MPI

**Pull request merged**